pkg/operation/botanist/etcdencryption.go

```diff
@@ -36,85 +38,197 @@ import (
+		}
+
+		if etcdSecretsEncryptionStatus == nil {
+			etcdSecretsEncryptionStatus := &encryptionconfiguration.ETCDEncryptionState{}
+			infodata.UpsertInfoData(&gardenerResourceDataList, common.ETCDSecretsEncryptionStateName, etcdSecretsEncryptionStatus)
+		}
+
+		shootState.Spec.Gardener = gardenerResourceDataList
+		return nil
+	})
+	if err != nil {
 		return err
 	}
+
+	b.ShootState = shootState
 	return nil
 }
 
-func (b *Botanist) createOrUpdateEncryptionConfiguration(ctx context.Context) (*corev1.Secret, error) {
+// ApplyEncryptionConfiguration creates or updates a secret on the Seed
+// which contains the encryption configuration that is necessary to encrypt the
+// Kubernetes secrets in etcd.
+func (b *Botanist) ApplyEncryptionConfiguration(ctx context.Context) error {
 	var (
 		secret = &corev1.Secret{ObjectMeta: kutil.ObjectMeta(b.Shoot.SeedNamespace, common.EtcdEncryptionSecretName)}
 		conf   *apiserverconfigv1.EncryptionConfiguration
 	)
 
 	_, err := controllerutil.CreateOrUpdate(ctx, b.K8sSeedClient.Client(), secret, func() error {
-		var err error
-		conf, err = encryptionconfiguration.ReadSecret(secret)
+		gardenerResourceList := gardencorev1alpha1helper.GardenerResourceDataList(b.ShootState.Spec.Gardener)
+		etcdSecretsEncryptionKeyInfoData, err := infodata.GetInfoData(gardenerResourceList, common.ETCDSecretsEncryptionKeyDataName)
 		if err != nil {
-			if !encryptionconfiguration.IsConfigurationNotFoundError(err) {
-				return err
-			}
-
-			b.Logger.Info("Creating new etcd encryption configuration for Shoot")
-			conf, err = encryptionconfiguration.NewPassiveConfiguration(time.Now(), rand.Reader)
-			if err != nil {
-				return err
-			}
+			return err
+		}
+		etcdSecretsEncryptionStateInfoData, err := infodata.GetInfoData(gardenerResourceList, common.ETCDSecretsEncryptionStateName)
+		if err != nil {
+			return err
 		}
 
-		// When firstly created, the encryption configuration secret does not have a checksum annotation yet. This annotation will
-		// only be added after all shoot secrets have been rewritten. In order to allow a smooth transition from un-encrypted to encrypted
-		// etcd data we first make the configuration inactive, i.e., put the `identity` provider as first list in the entry. In the next
-		// reconciliation we will detect that the annotation is set and then we can make it active, i.e., moving the `identity` provider
-		// to the second list item. The reason for this is that new API servers would otherwise start with an active configuration and would
-		// try to decrypt secrets in the etcd store (which would fail because they are not yet encrypted). Be aware that this will only happen
-		// once during the first introduction of the encryption configuration.
-		firstCreationOfEncryptionConfiguration := !metav1.HasAnnotation(secret.ObjectMeta, common.EtcdEncryptionChecksumAnnotationName)
```

Comment:
@plkokanov can you elaborate on this change? Where is this logic gone?
Actually, we discussed with @timuthy yesterday that we want to get rid of this anyways (we don't need to make the encryption inactive if the cluster is new - we can directly make it active). If I understand your changes correctly the logic is gone now, but what was your motivation to remove it?

Reply:
It's not gone, just handled differently: 
1. When we generate the key and state for the encryption config metadata, the state has `IsEncryptionActive` set to false.
2. When the encryption config for the secret is generated/deployed it checks the `IsEncryptionActive` and  creates it either in passive or active mode (first reconciliation will be passive)
3. Once the shoot secrets are rewritten `IsEncryptionActive` is changed to true
4. Then on next reconciliation the encryption config generation step will change the configuration to active.

To facilitate the new logic I can just remove the `isEncryptionActive` flag and only use `ForcePlainTextResources`

Reply:
I think it makes sense as you are anyways touching the code. @timuthy  WDYT?

Reply:
Yes, good suggestion. I would appreciate if you can do it in this PR @plkokanov.